ros2nodl/ros2nodl/verb/validate.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2020 Canonical Ltd.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License Version 3 as published by
+# the Free Software Foundation.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+from pathlib import Path
+import pprint
+import shutil
+import sys
+
+from argcomplete.completers import FilesCompleter
+import nodl
+from ros2cli.verb import VerbExtension
+
+
+class ValidateVerb(VerbExtension):
+    """Validate NoDL XML documents."""
+
+    def add_arguments(self, parser: argparse.ArgumentParser, cli_name: None = None):
+        command_group = parser.add_mutually_exclusive_group(required=True)
+        # Ignoring type because of https://github.com/python/typeshed/issues/1878
+        command_group.add_argument(  # type: ignore
+            'file', nargs='*', default=[], help='Specific .nodl.xml file(s) to validate.'
+        ).completer = FilesCompleter(allowednames=['nodl.xml'], directories=False)
+        command_group.add_argument(
+            '-a',
+            '--all',
+            action='store_true',
+            help='Validate all .nodl.xml files in the current directory.',
+        )
+
+        parser.add_argument('-p', '--print', action='store_true', help='Print parsed output.')
+
+    def main(self, args: argparse.Namespace) -> int:
+        paths = (
+            list(nodl._index._get_nodl_xml_files_in_path(path=Path.cwd()))
+            if args.all
+            else [Path(filename) for filename in args.file]
+        )
```

Comment:
This logic is hard to follow across multiple lines. Let's use a normal if/else here, please, which uses the same number of lines but is way more clear:

```suggestion
        if args.all:
            paths = nodl._index._get_nodl_xml_files_in_path(path=Path.cwd())
        else:
            paths = [Path(filename) for filename in args.file]
```

Reply:
fixed in 48641b3